webapp/src/App.tsx

```diff
@@ -143,10 +147,16 @@ const App: FC = () => {
                         <BackendProbe
                             uri={process.env.REACT_APP_BACKEND_URI as string}
                             onBackendFound={() => {
-                                setAppState(AppState.LoadingChats);
+                                setAppState(AppState.SettingUserInfo);
                             }}
                         />
                     )}
+                    {appState === AppState.SettingUserInfo &&
+                        (userInfoStatusText === defaultUserInfoStatusText ? (
```

Comment:
nit: is it a good practice to use string comparison rather than a flag to decide a state of the app?

Reply:
For example, we can have a flag `loadingUserInfoStatus`, with `true` being loading and `false` being failed. Then we can just pass the texts to `Loading` or `Error` instead of using `userInfoStatusText` in both places.

Reply:
You're so right. Since we're transitioning to another view altogether for the error state, I added another property in the AppState enum called ErrorLoadingUserInfo and that's what we transition to on error. This will help force the user to sign-out and sign back in as well.

Thanks for the callout!

Reply:
This is even better! Thanks!